lucet-runtime/lucet-runtime-internals/src/hostcall_macros.rs

```diff
@@ -0,0 +1,92 @@
+/// The macro that surrounds definitions of Lucet hostcalls in Rust.
+///
+/// It is important to use this macro for hostcalls, rather than exporting them directly, as it
+/// installs unwind protection that prevents panics from unwinding into the guest stack.
+///
+/// Since this is not yet a proc macro, the syntax is unfortunately fairly brittle. The functions it
+/// encloses must be of the form:
+///
+/// ```ignore
+/// #[$attr1]
+/// #[$attr2]
+/// ... // any number of attributes are supported; in most cases you will want `#[no_mangle]`
```

Comment:
What would be a case where would *not* want `#[no_mangle]`?

Reply:
It's useful when we're defining hostcalls for a `MockModuleBuilder`, so hostcalls for different tests don't interfere with each other.

Reply:
Actually, these aren't technically hostcalls, but are the fake guest functions that go into the module